<!--
  Arguments:
  lat   : Home latitude
  lon   : Home longitude
  alt   : Home altitude
  freq  : Update frequency of the PX4_msgs SensorGps message
  n_off : Offset angle (degrees) from geographical north to the vivo tracker calibrated y-axis.
           if drone is located directly south from ref lighthouse during calibration,
           the n_off shall be 0.0. If drone is west from lh, the n_off shall be 90.0
  decl  : Magnetic declination in the home global coordinates.
  mag   : Use magnetometer for calibration. 0: Not used, 1: Used.
           When enabling this, point the drone heading towards the reference lighthouse
           to calculate north_offset automatically.

  example usage:
    ros2 launch indoor_pos indoor_pos.launch lat:=60.000 lon:=31.000 alt:=50.0 n_off:=1.570796
    ros2 launch indoor_pos indoor_pos.launch lat:=60.000 lon:=31.000 alt:=50.0 mag:=1

-->

<launch>
  <arg name="lat" default="61.50341"/>
  <arg name="lon" default="23.77509"/>
  <arg name="alt" default="110.0"/>
  <arg name="freq" default="10"/>
  <arg name="n_off" default="0.0"/>
  <arg name="decl" default="0.0"/>
  <arg name="mag" default="0"/>

  <node name="indoor_pos" pkg="indoor_pos" exec="indoor_pos">
    <param name="home_lat" value="$(var lat)"/>
    <param name="home_lon" value="$(var lon)"/>
    <param name="home_alt" value="$(var alt)"/>
    <param name="frequency" value="$(var freq)"/>
    <param name="north_offset" value="$(var n_off)"/>
    <param name="declination" value="$(var decl)"/>
    <param name="use_mag" value="$(var mag)"/>
  </node>
</launch>

